File: Foundation/Number/ABAddition.cpp
#include<iostream>
using namespace std;

int anyBaseAddition(int n,  int m, int b)
{
    int res = 0, pow = 1, carry = 0;
    while(n != 0 || m != 0 || carry != 0)
    {
        int r1 = n % 10;
        int r2 = m % 10;
        n /= 10;
        m /= 10;
        int sum = r1 + r2 + carry;
        int r = sum % b;
        carry = sum / b;

        res += r * pow;
        pow *= 10;  
    }
    return res;
}

int main()
{
    int n, m, b;
    cin>>n>>m>>b;
    cout<<anyBaseAddition(n, m, b);
    return 0;
}

File: Foundation/Number/ABAddition.java
import java.util.Scanner;
public class ABAddition
{
    public static Scanner scn = new Scanner(System.in);
    public static int anyBaseAddition(int n1, int n2, int b)
    {
        int res = 0, pow = 1, carry = 0;
        while(n1 != 0 || n2 != 0 || carry != 0)
        {
            int r1 = n1 % 10;
            int r2 = n2 % 10;
            
            n1 /= 10;
            n2 /= 10;

            int sum = r1 + r2 + carry;
            int r = sum % b;
            carry = sum / b;

            res += r * pow;
            pow *= 10;
        }
        return res;
    }

    public static void main(String[] args)
    {
        int n1 = scn.nextInt();
        int n2 = scn.nextInt();
        int b = scn.nextInt();
        System.out.println(anyBaseAddition(n1, n2, b));
    }
}